JSIL.Libraries/Sources/JSIL.Core.js

```diff
@@ -3014,6 +3041,26 @@ JSIL.InstantiateProperties = function (publicInterface, typeObject) {
 $jsilcore.CanFixUpEnumInterfaces = false;
 
 JSIL.FixupInterfaces = function (publicInterface, typeObject) {
```

Comment:
Main changes is inside this method. It now calculates proper virtual member implementation for each base type, not only implemented interfaces.
Check is stricter now - method should be virtual and for auto-binding by name to interface class must explicitly declare that interface in it's own declaration.